Examples/Tests/MacroExamples/Implementation/ComplexMacros/DictionaryIndirectionMacroTests.swift

```diff
@@ -0,0 +1,108 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import MacroExamplesImplementation
+import SwiftSyntaxMacros
+import SwiftSyntaxMacrosTestSupport
+import XCTest
+
+final class DictionaryStorageMacroTests: XCTestCase {
+  private let macros: [String: Macro.Type] = [
+    "DictionaryStorage": DictionaryStorageMacro.self,
+    "DictionaryStorageProperty": DictionaryStoragePropertyMacro.self,
+  ]
+
+  func testExpansionConvertsStoredProperties() {
+    assertMacroExpansion(
+      """
+      @DictionaryStorage
+      struct Point {
+        var x: Int = 1
+        var y: Int = 2
+      }
+      """,
+      expandedSource: """
+        struct Point {
+          var x: Int = 1 {
+              get {
+                      _storage["x", default: 1] as! Int
+                }
+              set {
+                          _storage["x"] = newValue
+                }
+          }
+          var y: Int = 2 {
+              get {
+                      _storage["y", default: 2] as! Int
+                }
+              set {
+                          _storage["y"] = newValue
+                }
+          }
+
+            var _storage: [String: Any] = [:]
+        }
```

Comment:
Oh, this formatting is disgusting. You should be able to fix it by changing `DictionaryStoragePropertyMacro` to return

```swift
return [
  """
  get {
    _storage[\(literal: identifier.text), default: \(defaultValue)] as! \(type)
  }
  """,
  """
  set {
    _storage[\(literal: identifier.text)] = newValue
  }
  """,
]
```

Reply:
I've been thinking about doing it as another part of macros enhancements - I didn't want to change code without writing unit tests first but "this formatting is disgusting" is really good point so I've added all formatting related suggestion in a new commit after adding unit test: https://github.com/apple/swift-syntax/pull/2163/commits/39032ab805de40a7a4c7416a71a3f7704662a8ce

Reply:
Yeah, I thought the same about fixing it in a follow-up PR but the formatting was just too ugly for me. And since the fix is fairly straightforward, I thought we can fix it in this PR.